### PR TITLE
Rework snap directories

### PR DESCRIPTION
**Pull request by jardon**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
Directories in the snap are placed in random locations that are not intuitive or consistent with our other products.

* **What is the new behavior (if this is a feature change)?**
The directory layout for the snap mimics the original debian packaging which is what the associated rock also uses.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Yes.  Directory paths will need to be updated in the charm.

* **Other information**:
N/A

### CODE REVIEW

File: snap/hooks/install [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
-        
-echo "configuration folder ${CONFIG_DIR} does not exist."
-echo "copying default config to ${CONFIG_DIR}"
+SNAP_DATA=/var/snap/charmed-mysql/current
+MYSQL_CONFIG=$SNAP_DATA/etc/mysql/mysql.cnf
+MYSQLD_CONFIG=$SNAP_DATA/etc/mysql/mysql.conf.d/mysqld.cnf
+
+VAR_LIB=$SNAP_COMMON/var/lib/mysql
+ETC=$SNAP_DATA/etc/mysql
+VAR_RUN=$SNAP_COMMON/var/run/mysqld
```
Comment (shayancanonical) [inline snap/hooks/install:9]: should this be `$SNAP_COMMON/var/run/mysql`?

File: snap/hooks/install [hunk 2]
```diff
@@ -1,30 +1,38 @@
...
+VAR_LOG=$SNAP_COMMON/var/log/mysql
+RUN=$SNAP_COMMON/run/mysqld
```
Comment (shayancanonical) [inline snap/hooks/install]: i believe `/run` is a symlink to `/var/run`. so i'm not sure we need this variable (and can change the location of the mysqlx.sock file below to `/var/run`

File: snap/local/start-mysqld-exporter.sh [hunk 3]
```diff
@@ -2,5 +2,5 @@
 
 # For security measures, daemons should not be run as sudo. Execute mysqlrouter as the non-sudo user: snap-daemon.
 exec $SNAP/usr/bin/setpriv --clear-groups --reuid snap_daemon \
-  --regid snap_daemon -- $SNAP/bin/mysqld_exporter --config.my-cnf=$SNAP_COMMON/mysql/mysql.cnf "$@"
+  --regid snap_daemon -- $SNAP/bin/mysqld_exporter --config.my-cnf=$SNAP_COMMON/etc/mysql/mysql.cnf "$@"
```
Comment (shayancanonical) [inline snap/local/start-mysqld-exporter.sh]: should the config location be `$SNAP_DATA/etc/mysql/mysql.cnf`?

File: snap/local/start-mysqld.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
 exec $SNAP/usr/bin/setpriv --clear-groups --reuid snap_daemon \
-  --regid snap_daemon -- $SNAP/usr/bin/mysqld_safe --defaults-file=${SNAP_COMMON}/mysql/mysql.cnf "$@"
+  --regid snap_daemon -- $SNAP/usr/bin/mysqld_safe --defaults-file=$SNAP_DATA/etc/mysql/mysql.cnf "$@"
```
Comment (shayancanonical) [inline snap/local/start-mysqld.sh]: the group should be `root`?

File: snap/hooks/install (hunk 1 above)
Comment (jardon) [reply snap/hooks/install:9]: No, this is `var/run/mysqld` in the original debian packaging

File: snap/hooks/install (hunk 2 above)
Comment (jardon) [reply snap/hooks/install]: I moved this around for ya.  Good thought

**Review by shayancanonical — APPROVED** [review]

**Review by paulomach — APPROVED** [review]
lgtm

**Pull request merged**